ldms/src/ldmsd/ldmsd.c

```diff
@@ -1814,23 +1815,37 @@ int main(int argc, char *argv[])
 			break;
 		case 'A':
 			/* (multiple) auth options */
-			lval = strtok(optarg, "=");
+			
+			dup_auth = strdup(optarg);
+			if (!dup_auth) {
+				printf("Not enough memory parsing options!\n");
+				free(dup_auth);
```

Comment:
The very next line is exit(1). This just makes the arg parsing harder to read and bloats the code. I know you're trying to quiet valgrind, but at what cost?

Reply:
nothing to do with valgrind. just a mistake I will fix in freeing null unnecessarily. Certainly all calls to strdup should be checked for success, though.